### DISCUSSION TITLE
Installing only one component

### DISCUSSION THREAD

**Discussion by cysabi** [tailwindlabs/headlessui#1833] [General]
I noticed that in the documentation, you are given instructions to install the entire headless ui library.

Other headless accessible components, such as Radix Primitives & Reach UI are install-per-component, which I personally prefer.

Is there a way to install singular components in Headless UI? If not, I highly suggest at least giving users the option to install just one component. thank you!

**Comment by RobinMalfait** [comment]
It's not possible to install a single package only, however everything is codesplittable (if that's a word) so if you only use a single component, you will only get that component in your bundle.

If you are worried about file size in your node_modules folder, Headless UI doesn't have any dependencies.

**Reply by cysabi** [reply]
Thanks for the reply, my main reason is just ergonomics and convention. It just feels slightly more unnatural to install the entire bundle after I'm used to installing singular components from reach or radix

I totally understand this reason being completely superficial and probably not worth changing, I just wanted to bring up how other headless ui frameworks install.

**Reply by moaazbhnas228** [reply]
Actually, I'm only using `Menu`, and I find in the compiled code references to other components.
